src/compiler/crystal/syntax/lexer.cr

```diff
@@ -3178,11 +3183,14 @@ module Crystal
     end
 
     def self.ident_start?(char)
-      char.ascii_letter? || char == '_' || char.ord > 0x9F
+      char.letter? || char == '_' || Unicode.number_letter?(char)
```

Comment:
I don't think it make sense to add numbers in other formats as identifiers. Be it Ⅷ or 8, they represent the same thing.

```suggestion
      char.letter? || char == '_'
```